distributed/tests/test_worker.py

```diff
@@ -1044,11 +1044,22 @@ async def test_pid(s, a, b):
 
 
 @gen_cluster(client=True)
-async def test_get_client(c, s, a, b):
+@pytest.mark.parametrize("asynchronous", (True, False))
+async def test_get_client(c, s, a, b, asynchronous):
     def f(x):
-        cc = get_client()
-        future = cc.submit(inc, x)
-        return future.result()
+        cc = get_client(asynchronous=asynchronous)
+        assert cc._asynchronous == asynchronous
+
+        if asynchronous:
+
+            async def _():
+                future = cc.submit(inc, x)
+                return await future.result()
+
+            return cc.sync(_, asynchronous=False)
```

Comment:
I find this a bit confusing due to the `asynchronous` kwarg. I might suggest to be more explicit by not using the Client.sync method but the raw sync function instead.

```suggestion
            from distributed.utils import sync
            return sync(cc.loop, _)
```

Reply:
Confused me as well, using `sync` is easier on the mind. :+1: https://github.com/dask/distributed/pull/7871/commits/4fbb8ef935aaa0b5a88a69966c7cf73cdeb26bb9